File: apps/otel-go/handlers/health.go
// SPDX-FileCopyrightText: Copyright (C) Nicolas Lamirault <nicolas.lamirault@gmail.com>
// SPDX-License-Identifier: Apache-2.0

package handlers

import (
	"net/http"

	"github.com/gin-gonic/gin"
	"go.opentelemetry.io/otel/trace"
	"go.uber.org/zap"
)

type HealthHandler struct{}

func NewHealthHandler() *HealthHandler {
	return &HealthHandler{}
}

func (h *HealthHandler) Health(ctx *gin.Context) {
	span := trace.SpanFromContext(ctx)
	zap.L().Info("Health status",
		zap.String("trace_id", span.SpanContext().TraceID().String()),
		zap.String("span_id", span.SpanContext().SpanID().String()))
	ctx.String(http.StatusOK, "ok")
}

File: apps/otel-rust/src/routes/health.rs
// SPDX-FileCopyrightText: Copyright (C) Nicolas Lamirault <nicolas.lamirault@gmail.com>
// SPDX-License-Identifier: Apache-2.0

use axum::Json;
use serde_json::{json, Value};
use tracing::info;

pub async fn handler_health() -> Json<Value> {
    info!(target: "otel_rust::handlers", "Health handler");
    Json(json!({ "status": "ok" }))
}